tests/python_packages.toml

```diff
@@ -17,3 +17,7 @@ sha256 = "b305b30c99526df5e63d6022dd2310a0a941a187bd3884f4c8ef0418df6c39f3"
 [click]
 url = "https://files.pythonhosted.org/packages/59/87/84326af34517fca8c58418d148f2403df25303e02736832403587318e9e8/click-8.1.3.tar.gz"
 sha256 = "7682dc8afb30297001674575ea00d1814d808d6a36af415a82bd481d37ba7b8e"
+
+[leftpadx-wheel]
+url = "https://files.pythonhosted.org/packages/37/ca/41e9d0964799b00ad00c6254e6c0bc1a26c604456ab659e13cf91e5bdd81/leftpadx-0.0.1-py3-none-any.whl"
+sha256 = "512084949099fab88042be250a6da55dcf933ac3af8feffddefd52dace66160c"
```

Comment:
Ah, ok. In general I suspect that any `*-py3-none-any.whl` should be fine to cache instead of the corresponding `*.tar.gz`, as we should never run into platform/version incompatibilities for those.

I agree with your disappointment that installation is not much improved by using wheels instead of tarball. Maybe the overhead of creating the venv itself is too large in comparison? Looking at `test_local_env`, we have two tests that create a new venv with `pip` inside:
- `test_local_env__default_venv__contains_pip_and_setuptools` (creates the venv in the test itself)
- `test_on_installed_venv__returns_local_deps` (creates the venv inside `TemporaryPipInstallResolver`)
and they each take 2-3s to run on my machine. All the other `test_local_env` tests are <0.05s.

I suspect this ends up supporting the argument in favor of caching the results of `install_deps`, so that these venvs can be used as little as possible. Won't help much with test coverage of this code path, though... 😿

Reply:
Ah, sorry, I was only referring to the cache mentioned here:

![fawlty-deps-pypi-realm](https://user-images.githubusercontent.com/547031/234072889-b8259c76-cad7-4192-9c03-9a8a92efb0fb.png)

With something like that in place, the user should most of the time not have to wait for FD to create venvs and install deps.

But that does not help us now, precisely because we _want_ to test that creating the venv and installing the deps actually works as it should.

Or at least: we want to test this in the `TemporaryPipInstallResolver` unit tests. The `test_resolver` PBT tests should maybe be allowed to instrument/mock this part of `TemporaryPipInstallResolver` to speed things up? Maybe using a single venv across the entire `test_resolver` session (so as to avoid the setup cost for venvs) is sufficient to make the tests fast enough?